### ISSUE TITLE
openwakeword error after some time. Voice Assist speakers stop responding. 

### ISSUE THREAD

**Issue by stang70man** [home-assistant/addons#3484]
### Describe the issue you are experiencing

I am finding an error with the openwakeword addon after some indeterminate time. (Usually within 24 hours or so.) Restarting the openwakeword service will bring the voice assistants back online until I receive the error again.

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

CEC Scanner

### What is the version of the add-on?

Openwakeword v1.8.2

### Steps to reproduce the issue

1. Restart openwakeword service. It runs for several hours to a day
2. Virtual Assistants become unresponsive
3. Restart openwakeword service. It runs again for several hours to a day.
...

## System Information

version | core-2024.2.2
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.1
os_name | Linux
os_version | 6.1.74-haos
arch | x86_64
timezone | America/Chicago
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.34.0
Stage | running
Available Repositories | 1404
Downloaded Repositories | 44
HACS Data | ok

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | January 31, 2025 at 6:00 PM
relayer_connected | true
relayer_region | us-east-1
remote_enabled | true
remote_connected | true
alexa_enabled | true
google_enabled | true
remote_server | us-east-1-1.ui.nabu.casa
certificate_status | ready
instance_id | 18ace9f2c705453a8c19e89cdc4deaec
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>EyeOnWater</summary>

api_endpoint_reachable | ok
-- | --

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 11.5
-- | --
update_channel | stable
supervisor_version | supervisor-2024.02.0
agent_version | 1.6.0
docker_version | 24.0.7
disk_total | 156.8 GB
disk_used | 36.1 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Samba share (12.3.0), Terminal & SSH (9.9.0), UniFi Network Application (3.0.3), File editor (5.8.0), Home Assistant Google Drive Backup (0.112.1), Network UPS Tools (0.13.0), Zigbee2MQTT (1.35.3-1), RTSPtoWeb - WebRTC (1.4.0), Tailscale (0.16.0), Whisper (1.0.2), Piper (1.4.0), openWakeWord (1.8.2), ESPHome (2024.2.0), Studio Code Server (5.15.0)

</details>

<details><summary>Dashboards</summary>

dashboards | 3
-- | --
resources | 30
views | 36
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | February 21, 2024 at 3:03 AM
-- | --
current_recorder_run | February 21, 2024 at 8:21 PM
estimated_db_size | 811.78 MiB
database_engine | sqlite
database_version | 3.44.2

</details>

<details><summary>Spotify</summary>

api_endpoint_reachable | ok
-- | --

</details>

### Anything in the Supervisor logs that might be useful for us?

```txt
[32m24-02-23 18:01:41 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:01:41 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:01:54 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:01:54 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:01:55 INFO (MainThread) [supervisor.api.middleware.security] /backups/5fac790c/download access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:01:55 INFO (MainThread) [supervisor.api.backups] Downloading backup 5fac790c[0m
[32m24-02-23 18:01:55 INFO (MainThread) [supervisor.api.middleware.security] /backups/5fac790c/download access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:01:55 INFO (MainThread) [supervisor.api.backups] Downloading backup 5fac790c[0m
[32m24-02-23 18:02:13 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:02:13 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[33m24-02-23 18:03:05 WARNING (MainThread) [supervisor.homeassistant.websocket] Connection is closed[0m
[32m24-02-23 18:03:10 INFO (MainThread) [supervisor.addons.addon] Starting add-on a0d7b954_unifi again[0m
[33m24-02-23 18:03:10 WARNING (MainThread) [supervisor.addons.options] Option 'ssl' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:03:10 WARNING (MainThread) [supervisor.addons.options] Option 'certfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:03:10 WARNING (MainThread) [supervisor.addons.options] Option 'keyfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:03:10 WARNING (MainThread) [supervisor.addons.options] Option 'ssl' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:03:10 WARNING (MainThread) [supervisor.addons.options] Option 'certfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:03:10 WARNING (MainThread) [supervisor.addons.options] Option 'keyfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[32m24-02-23 18:03:15 INFO (MainThread) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/unifi/amd64 with version 3.0.3[0m
[32m24-02-23 18:03:15 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon a0d7b954_unifi[0m
[32m24-02-23 18:03:15 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on core_configurator[0m
[32m24-02-23 18:03:15 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon core_configurator[0m
[32m24-02-23 18:03:15 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:03:16 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:03:16 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on a0d7b954_nut[0m
[32m24-02-23 18:03:16 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon a0d7b954_nut[0m
[32m24-02-23 18:03:17 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on 45df7312_zigbee2mqtt[0m
[32m24-02-23 18:03:17 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon 45df7312_zigbee2mqtt[0m
[32m24-02-23 18:03:18 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on 43647439_rtsp-to-web[0m
[32m24-02-23 18:03:18 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon 43647439_rtsp-to-web[0m
[32m24-02-23 18:03:18 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on a0d7b954_tailscale[0m
[32m24-02-23 18:03:18 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon a0d7b954_tailscale[0m
[32m24-02-23 18:03:18 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on core_whisper[0m
[32m24-02-23 18:03:18 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon core_whisper[0m
[32m24-02-23 18:03:19 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on core_piper[0m
[32m24-02-23 18:03:19 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon core_piper[0m
[32m24-02-23 18:03:19 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on core_openwakeword[0m
[32m24-02-23 18:03:19 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon core_openwakeword[0m
[32m24-02-23 18:03:19 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on 5c53de3b_esphome[0m
[32m24-02-23 18:03:19 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon 5c53de3b_esphome[0m
[32m24-02-23 18:03:19 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on a0d7b954_vscode[0m
[33m24-02-23 18:03:19 WARNING (MainThread) [supervisor.addons.options] Option 'ssl' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:03:19 WARNING (MainThread) [supervisor.addons.options] Option 'certfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:03:19 WARNING (MainThread) [supervisor.addons.options] Option 'keyfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[32m24-02-23 18:03:20 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon a0d7b954_vscode[0m
[32m24-02-23 18:03:20 INFO (MainThread) [supervisor.backups.manager] Backup 253c960c starting stage home_assistant[0m
[32m24-02-23 18:03:20 INFO (MainThread) [supervisor.homeassistant.module] Backing up Home Assistant Core config folder[0m
[33m24-02-23 18:03:20 WARNING (MainThread) [supervisor.addons.options] Option 'ssl' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:03:20 WARNING (MainThread) [supervisor.addons.options] Option 'certfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:03:20 WARNING (MainThread) [supervisor.addons.options] Option 'keyfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[32m24-02-23 18:03:49 INFO (MainThread) [supervisor.homeassistant.module] Backup Home Assistant Core config folder done[0m
[32m24-02-23 18:03:49 INFO (MainThread) [supervisor.backups.manager] Backup 253c960c starting stage folders[0m
[32m24-02-23 18:03:49 INFO (SyncWorker_1) [supervisor.backups.backup] Backing up folder share[0m
[32m24-02-23 18:03:49 INFO (SyncWorker_1) [supervisor.backups.backup] Backup folder share done[0m
[32m24-02-23 18:03:49 INFO (SyncWorker_0) [supervisor.backups.backup] Backing up folder addons/local[0m
[32m24-02-23 18:03:49 INFO (SyncWorker_0) [supervisor.backups.backup] Backup folder addons/local done[0m
[32m24-02-23 18:03:49 INFO (SyncWorker_2) [supervisor.backups.backup] Backing up folder ssl[0m
[32m24-02-23 18:03:49 INFO (SyncWorker_2) [supervisor.backups.backup] Backup folder ssl done[0m
[32m24-02-23 18:03:49 INFO (SyncWorker_4) [supervisor.backups.backup] Backing up folder media[0m
[32m24-02-23 18:04:11 INFO (SyncWorker_4) [supervisor.backups.backup] Backup folder media done[0m
[32m24-02-23 18:04:11 INFO (MainThread) [supervisor.backups.manager] Backup 253c960c starting stage finishing_file[0m
[32m24-02-23 18:04:11 INFO (MainThread) [supervisor.backups.manager] Backup 253c960c starting stage await_addon_restarts[0m
[32m24-02-23 18:04:21 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:21 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:21 INFO (MainThread) [supervisor.api.middleware.security] /backups/253c960c/info access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:21 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:21 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:22 INFO (MainThread) [supervisor.api.middleware.security] /backups/3f28ab6f access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:30 INFO (MainThread) [supervisor.backups.manager] Removed backup file 3f28ab6f[0m
[32m24-02-23 18:04:30 INFO (MainThread) [supervisor.api.middleware.security] /backups/253c960c/download access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:30 INFO (MainThread) [supervisor.api.backups] Downloading backup 253c960c[0m
[32m24-02-23 18:04:50 INFO (MainThread) [supervisor.backups.manager] Creating full backup with slug 253c960c completed[0m
[32m24-02-23 18:04:52 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:52 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:52 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:52 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:53 INFO (MainThread) [supervisor.api.middleware.security] /backups/253c960c/download access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:53 INFO (MainThread) [supervisor.api.backups] Downloading backup 253c960c[0m
[32m24-02-23 18:04:53 INFO (MainThread) [supervisor.api.middleware.security] /backups/253c960c/download access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:04:53 INFO (MainThread) [supervisor.api.backups] Downloading backup 253c960c[0m
[33m24-02-23 18:04:57 WARNING (MainThread) [supervisor.addons.options] Option 'interface' does not exist in the schema for Samba share (core_samba)[0m
[33m24-02-23 18:04:57 WARNING (MainThread) [supervisor.addons.options] Option 'ssl' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:04:57 WARNING (MainThread) [supervisor.addons.options] Option 'certfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:04:57 WARNING (MainThread) [supervisor.addons.options] Option 'keyfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:04:57 WARNING (MainThread) [supervisor.addons.options] Option 'uspd_maxage' does not exist in the schema for Network UPS Tools (a0d7b954_nut)[0m
[33m24-02-23 18:04:57 WARNING (MainThread) [supervisor.addons.options] Option 'uspmon_deadtime' does not exist in the schema for Network UPS Tools (a0d7b954_nut)[0m
[32m24-02-23 18:06:15 INFO (MainThread) [supervisor.api.middleware.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m24-02-23 18:06:15 INFO (MainThread) [supervisor.api.middleware.security] /backups access from cebe7a76_hassio_google_drive_backup[0m
[33m24-02-23 18:09:57 WARNING (MainThread) [supervisor.addons.options] Option 'interface' does not exist in the schema for Samba share (core_samba)[0m
[33m24-02-23 18:09:57 WARNING (MainThread) [supervisor.addons.options] Option 'ssl' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:09:57 WARNING (MainThread) [supervisor.addons.options] Option 'certfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:09:57 WARNING (MainThread) [supervisor.addons.options] Option 'keyfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:09:57 WARNING (MainThread) [supervisor.addons.options] Option 'uspd_maxage' does not exist in the schema for Network UPS Tools (a0d7b954_nut)[0m
[33m24-02-23 18:09:57 WARNING (MainThread) [supervisor.addons.options] Option 'uspmon_deadtime' does not exist in the schema for Network UPS Tools (a0d7b954_nut)[0m
[33m24-02-23 18:14:57 WARNING (MainThread) [supervisor.addons.options] Option 'interface' does not exist in the schema for Samba share (core_samba)[0m
[33m24-02-23 18:14:57 WARNING (MainThread) [supervisor.addons.options] Option 'ssl' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:14:57 WARNING (MainThread) [supervisor.addons.options] Option 'certfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:14:57 WARNING (MainThread) [supervisor.addons.options] Option 'keyfile' does not exist in the schema for UniFi Network Application (a0d7b954_unifi)[0m
[33m24-02-23 18:14:57 WARNING (MainThread) [supervisor.addons.options] Option 'uspd_maxage' does not exist in the schema for Network UPS Tools (a0d7b954_nut)[0m
[33m24-02-23 18:14:57 WARNING (MainThread) [supervisor.addons.options] Option 'uspmon_deadtime' does not exist in the schema for Network UPS Tools (a0d7b954_nut)[0m
```

### Anything in the add-on logs that might be useful for us?

```txt
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service openwakeword: starting
s6-rc: info: service openwakeword successfully started
s6-rc: info: service discovery: starting
INFO:root:Ready
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
[06:49:31] INFO: Successfully sent discovery information to Home Assistant.
s6-rc: info: service discovery successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
ERROR:root:Unexpected error in embeddings thread
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/dist-packages/wyoming_openwakeword/openwakeword.py", line 167, in embeddings_proc
    mels_tensor[i, :, :, 0] = client.mels[
    ~~~~~~~~~~~^^^^^^^^^^^^
ValueError: could not broadcast input array from shape (970,32) into shape (76,32)
```

**Comment by micseydel** [comment]
If this is the [same bug,](https://github.com/home-assistant/addons/issues/3464) it sounds like it's working better for you than most of us 🙃

**Comment by stang70man** [comment]
> If this is the [same bug,](https://github.com/home-assistant/addons/issues/3464) it sounds like it's working better for you than most of us 🙃

I had this issue early on when I was running Proxmox on an old Intel Dual Core processor with limited memory. I replaced my hardware with an Intel i5 Gen6 processor 6 or 8 cores and with 32Gb of memory and threw more memory at the VM and that helped it a bunch. This was an upgrade I needed to do anyway, but performance on wake word pushed me to go ahead. Not saying that will work for anyone else, but it certainly solved my earlier issues.
